app/services/foreman_ansible/operating_system_parser.rb

```diff
@@ -0,0 +1,102 @@
+# frozen_string_literal: true
+
+module ForemanAnsible
+  # Methods to parse facts related to the OS
+  module OperatingSystemParser
+    def operatingsystem
+      args = { :name => os_name, :major => os_major, :minor => os_minor }
+      args[:release_name] = os_release_name if os_name == 'Debian' || os_name == 'Ubuntu'
+      return @local_os if local_os(args).present?
+      return @new_os if new_os(args).present?
+      logger.debug do
+        'Ansible facts parser: No OS could be created with '\
+        "os_name='#{os_name}' os_major='#{os_major}' "\
+        "os_minor='#{os_minor}': "\
+        "#{@new_os.errors if @new_os.present?}"
+      end
+      nil
+    end
+
+    def local_os(args)
+      @local_os = Operatingsystem.where(args).first
+    end
+
+    def new_os(args)
+      return @new_os if @new_os.present?
+      @new_os = Operatingsystem.new(args.merge(:description => os_description))
+      @new_os if @new_os.valid? && @new_os.save
+    end
+
+    def debian_os_major_sid
+      case facts[:ansible_distribution_major_version]
+      when /wheezy/i
+        '7'
+      when /jessie/i
+        '8'
+      when /stretch/i
+        '9'
+      when /buster/i
+        '10'
+      end
```

Comment:
This could be moved into operatingsystems/debian.rb at some point, do not do that now but make a note.